internal/internal_event_handlers.go

```diff
@@ -985,8 +984,14 @@ func (weh *workflowExecutionEventHandlerImpl) handleActivityTaskFailed(event *ev
 	}
 
 	attributes := event.GetActivityTaskFailedEventAttributes()
-	err := constructError(attributes.Reason, attributes.Details, weh.GetDataConverter())
-	activity.handle(nil, err)
+	activityTaskErr := NewActivityTaskError(
+		attributes.GetScheduledEventId(),
+		attributes.GetStartedEventId(),
+		attributes.GetIdentity(),
+		convertFailureToError(attributes.GetFailure(), weh.GetDataConverter()),
+	)
```

Comment:
Here and below, actual error is wrapped inside `NewActivityTaskError`/`NewChildWorkflowExecutionError`.